Assignment3/PointAndGrid.py

```diff
@@ -0,0 +1,260 @@
+class Point:
+
+    def __init__(self, grid = None, coordinates = None, value = None):
+        """
+        For Point initialization with optional grid and coordinates.
+        :param grid: Grid
+        :param coordinates: Tuple(int, int)
+        :return: None
+        """
+        self.grid = grid
+        self.coordinates = coordinates
+        self.value = value
+        if grid and coordinates:
+            v = grid.getElement(coordinates)
+            if value and value != v:
+                raise AttributeError
+            else: self.value = v
+
+    def setGrid(self, gr):
+        """
+        :param gr: Grid
+        """
+        self.grid = gr
+
+    def setCoordinates(self, co):
+        """
+        :param co: Tuple(int, int)
+        """
+        self.coordinates = co
+
+    def setValue(self, va):
+        """
+        :param va: any
+        """
+        self.value = va
+
+    def __str__(self):
+        """
+        For string representation of Point.
+        :return: string
+        """
+        return str(self.value)
+
+    def __eq__(self, other):
+        """
+        Supports equivalence for the Point. Returns True if the elements self and other are equal, False otherwise.
+        :param other: any type
+        :return: bool
+        """
+        if isinstance(other, self.__class__):
+            return self.grid == other.grid and self.coordinates == other.coordinates and self.value == other.value
+        else:
+            return False
+
+class Grid:
+
+    def __init__(self, grid):
+        """
+        For Grid intialization with a grid of elements.
+        :param grid: list(list())
+        :return: None
+        """
+        self.grid = grid
+
+    def horysontalSize(self):
+        """
+        Returns number of columns. Size of a row.
+        :return: int
+        """
+        return len(self.grid[0]) if self.grid is not None else 0
+
+    def verticalSize(self):
+        """
+        Returns number of rows. Size of a column.
+        :return: int
+        """
+        return len(self.grid) if self.grid is not None else 0
+
+    def addRow(self, row):

```

Comment:
OK :) 
I know it is usefulness here. But that are great functions and it was interesting to implement them! But sure, I will delete them as Magdalena will do the review too and will take it into account for the next times, thanks.